homeassistant/components/sensor/magicseaweed.py

```diff
@@ -0,0 +1,212 @@
+"""
+Support for magicseaweed data from magicseaweed.com.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.magicseaweed/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, CONF_NAME, CONF_MONITORED_CONDITIONS, ATTR_ATTRIBUTION)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['magicseaweed==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOURS = 'hours'
+CONF_SPOT_ID = 'spot_id'
+CONF_UNITS = 'units'
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+DEFAULT_UNIT = 'us'
+DEFAULT_NAME = 'MSW'
+DEFAULT_ATTRIBUTION = "Data provided by magicseaweed.com"
+
+ICON = 'mdi:waves'
+
+API_URL = 'http://magicseaweed.com/api/{}/forecast/'
+
+HOURS = ['12AM', '3AM', '6AM', '9AM', '12PM', '3PM', '6PM', '9PM']
+
+SENSOR_TYPES = {
+    'max_breaking_swell': ['Max'],
+    'min_breaking_swell': ['Min'],
+    'swell_forecast': ['Forecast'],
+}
+
+UNITS = ['eu', 'uk', 'us']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SPOT_ID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOURS, default=None):
+        vol.All(cv.ensure_list, [vol.In(HOURS)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNITS): vol.In(UNITS),
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=300)): (
```

Comment:
This seems very aggressive. We should also not make it configurable and just pick the right amount. We don't want Home Assistant to be used to hammer a public API.